src/main/java/com/project/stockexchangeappbackend/rest/UsersController.java

```diff
@@ -78,6 +80,29 @@ public UserDTO getUser(@ApiParam(value = "Id of desired user.", required = true)
                 .map(user -> mapper.map(user, UserDTO.class));
     }
 
+    @GetMapping("/config/user-data")
+    @PreAuthorize("hasRole('USER')")
+    @ApiOperation(value = "Retrieve logged in user", response = UserDTO.class, notes = "Required role: USER")
+    @ApiResponses({@ApiResponse(code = 200, message = "User was successfully retrieved."),
+            @ApiResponse(code = 404, message = "Given user not found.", response = ErrorResponse.class)})
+    public UserDTO getUser(Principal principal) {
+        return mapper.map(userService.findUserByEmail(principal.getName()), UserDTO.class);
+    }
+
+    @PostMapping("/config/change-password")
+    @PreAuthorize("hasRole('USER')")
+    @ApiOperation(value = "Change logged in user password", response = UserDTO.class, notes = "Required role: USER")
+    @ApiResponses({@ApiResponse(code = 200, message = "User password was successfully changed."),
+            @ApiResponse(code = 404, message = "Failed to change user password.", response = ErrorResponse.class)})
+    @ApiImplicitParams({
+            @ApiImplicitParam(name = "password", dataType = "string", paramType = "body",
+                    value = "New password"),
+    })
+    public UserDTO changePassword(@RequestParam String password, Principal principal) {
```

Comment:
co jeśli password będzie null albo " "? To nie może tak działać. Należy stworzyć nowy obiekt DTO, który będzie zawierał hasło, które będzie spełniało kryteria zdefiniowane podczas rejestracji. Myślę że należy rozważyć również obowiązek podania starego hasła podczas rejestracji

Reply:
Dodatkowo nie musisz zwracać UserDTO. To nie jest błąd, ale przyjeliśmy zasadę że tylko GET zwraca dane. Status wykonania operacji jest raportowany przy pomocy status code

Reply:
Done.